WHITEPAPER.md

```diff
@@ -1,6 +1,47 @@
-# Anon Inu Whitepaper
+# Lily Whitepaper
 
-# Abstract
 
 # 1. Introduction
 
+Lilynet is a decentralized trustless minimal-knowledge network that acts as a relay between two communicating parties via HTTP/HTTPS protocols. Specifically, when a user attempts to send a request to some website, the information can be intercepted and analyzed by various third parties (nefarious, Orwellian, etc.). Projects like <cite>[Tor][1]</cite> and <cite>[I2P][2]</cite> attempt to obfuscate the path between a user and his/her destination. In standard TCP/IP communication, both parties have information (specifically the IP address) of the other party. While Tor relay nodes remove this information, the problem is the lack of nodes due partially to the lack of incentive of being a relay. The figure below shows that there are currently 6,000 Tor relays and they are maintained in a <cite>[centralized list][3]</cite>, which makes it trivial for nefarious actors to block the services.
+
+[<img src="https://user-images.githubusercontent.com/1019677/143939802-29eda65e-35a1-4d66-9c75-eea0a03fc409.png" width="50%"/>](https://user-images.githubusercontent.com/1019677/143939802-29eda65e-35a1-4d66-9c75-eea0a03fc409.png)
+
+Lilynet aims to solve this problem by adhering to 3 principles:
+
+1. The network should be completely decentralized
+2. The market determines the price of privacy
+3. The third
+
+
+## 1.1 Definitions
+To keep consistency in this proposal, the following definitions will be referred to:
+- `visitor node` - the computer of an entity that wishes to access some endpoint
+- `bandwidth` - the data transfer capacity of a computer network in bits per second (Bps)
+- `throughput` - the actual amount of data that is successfully sent/received over the communication link, measured in bits per second (Bps)
+- `relay node` - a computer in the network that is relaying information between 2 parties
+- `peer` - a `relay node` that either a `visitor` or `relay node` is connected to
+- `coin` - a unit of monetary value. Details TBD
+
+
+# 2. Decentralized Network Discovery
+
+Each `node` in the network is connected (by default) to 10 `peers`. To boostrap the list, similar to the <cite>[Bitcoin P2P bootstrapping mechanism][4]</cite>, Bridgenet will first try to connect to peers that it has connected to before, saved in its local database. If no cached peers can be connected to, it will query a DNS seed to get a list of active peers. If none of those are reachable, it will fallback to a hardcoded peer address. Once a peer is reached, it will be queried to find additional peers.
+
+Each node will send a heartbeat packet (of 10MB) to measure `throughput` periodically. Any peers that have a throughput slower than 1 standard deviation away from the average throughput will be dropped and new peers will be sought.
```

Comment:
Should we let the peers know they've been dropped to give them the opportunity to fix their bandwidth ?

Reply:
Hmmm - I’m not 100% sure, but I think they’ll know, right?

If `A` has peers, `B`, `C`, etc, that means at least one of `B`’s peers is `A`, right? So when `A` drops `B`, then when `B` sends a heartbeat to `A`, it’ll error.

What do you think @kfaria ?

Reply:
@kfaria - do you think I should include this `mutual peer concept in the whitepaper?